src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/MethodCodeNode.cs

```diff
@@ -99,6 +99,16 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
                     dependencies.Add(new DependencyListEntry(factory.MethodEntrypoint(invokeStub), "Reflection invoke"));
             }
 
+            if (_method.HasInstantiation)
```

Comment:
This doesn't look right. Should this actually be a call to `IsMethodEligibleForTracking`?